cfme/utils/providers.py

```diff
@@ -387,6 +387,9 @@ def get_mgmt(provider_key, providers=None, credentials=None):
         provider_kwargs['provider_key'] = provider_key
     provider_kwargs['logger'] = logger
 
+    if isinstance(provider_key, Mapping):
```

Comment:
I'm still not confident in this change, and I think we need to look at approaching this from how the function is called and how he kwargs are used.

This exact check of the type of provider_key is already done at L359, setting provider_key to the 'name' attribute of the provider_data dictionary.  I'm not sure why that isn't functional in your case.

If you're trying to create a mgmt object for something that's not in cfme_data.yaml, then the custom dictionary should be passed as the second argument, `providers`.